### DISCUSSION TITLE
Replication placement per disk instead of per node when using **1 or **2 replication for read performance

### DISCUSSION THREAD

**Discussion by tobiasmuehl** [seaweedfs/seaweedfs#2973] [Q&A] [answered]
I have volume servers with multiple disks. On some servers, I run multiple volume servers, one per disk. On others, I run only only one volume server with multiple `-dir` params (e.g. `-dir=/weed,/weed2,/weed3,/weed4`. 

I'm trying to confirm whether I can add same-rack replicas to increase read performance (e.g. from `010` to `012`). I traced the source code all the way to the [`location`](https://sourcegraph.com/github.com/chrislusf/seaweedfs@3bce568c32f78a5bf12f0a31995d4098ca97fc6b/-/blob/weed/shell/command_volume_fix_replication.go?L442:6) type, which contains

```go
type location struct {
  dc       string
  rack     string
  dataNode *master_pb.DataNodeInfo
  }
```

In the same file there is [`avoid duplicated volume on the same data node`](https://sourcegraph.com/github.com/chrislusf/seaweedfs@3bce568c32f78a5bf12f0a31995d4098ca97fc6b/-/blob/weed/shell/command_volume_fix_replication.go?L351). Looks to me that replication is only considered on a _per node_ level, not a _per disk_ level. So, in order to increase read performance, I might need to run one volume server per disk, instead of one volume server per physical node. Am I correct?

Example topology: 1 DC with 3 racks with 1 node each, each node with 4 disks. Wanted replication `022`.

**Comment by chrislusf** [comment, marked as answer]
> I might need to run one volume server per disk, instead of one volume server per physical node. Am I correct?

Correct.
